src/any_llm/any_llm.py

```diff
@@ -147,6 +152,24 @@ def _create_provider(
             raise ImportError(msg) from e
 
         provider_class: type[AnyLLM] = getattr(module, provider_class_name)
+
+        if api_key:
```

Comment:
I think at this point we should also check for `os.environ[cls.ANY_LLM_KEY]`, so user setting that env var also serves as activation of the platform.

```suggestion
        if api_key:
```